core/drivers/crypto/caam/hal/common/hal_ctrl.c

```diff
@@ -10,50 +10,82 @@
 #include <registers/ctrl_regs.h>
 #include <registers/version_regs.h>
 
-uint8_t caam_hal_ctrl_jrnum(vaddr_t baseaddr)
+uint8_t caam_hal_ctrl_era(vaddr_t baseaddr)
 {
 	uint32_t val = 0;
 
-	val = io_caam_read32(baseaddr + CHANUM_MS);
+	/* Read the number of instance */
+	val = io_caam_read32(baseaddr + CCBVID);
 
-	return GET_CHANUM_MS_JRNUM(val);
+	return GET_CCBVID_CAAM_ERA(val);
 }
 
-uint8_t caam_hal_ctrl_hash_limit(vaddr_t baseaddr)
+uint8_t caam_hal_ctrl_jrnum(vaddr_t baseaddr)
 {
 	uint32_t val = 0;
+	uint8_t jrnum;
 
-	/* Read the number of instance */
-	val = io_caam_read32(baseaddr + CHANUM_LS);
+	if (caam_hal_ctrl_era(baseaddr) < 10) {
+		val = io_caam_read32(baseaddr + CHANUM_MS);
 
-	if (GET_CHANUM_LS_MDNUM(val)) {
-		/* Hashing is supported */
-		val = io_caam_read32(baseaddr + CHAVID_LS);
-		val &= BM_CHAVID_LS_MDVID;
-		if (val == CHAVID_LS_MDVID_LP256)
-			return TEE_MAIN_ALGO_SHA256;
+		jrnum = GET_CHANUM_MS_JRNUM(val);
+	} else {
+		val = io_caam_read32(baseaddr + JR_VERSION);
 
-		return TEE_MAIN_ALGO_SHA512;
+		jrnum = GET_JR_VERSION_JRNUM(val);
 	}
 
-	return UINT8_MAX;
+	return jrnum;
```

Comment:
You could do ```return GET_CHANUM_MS_JRNUM(val);``` and ```return GET_JR_VERSION_JRNUM(val);``` so you don't need ```uint8_t jrnum;```
Up to you

Reply:
I did that earlier and then changes it..I will revert to return from if else loop and avoid the new variable

Reply:
changing it to return gives checkpatch error, so I have reverted it to variable approach.

Reply:
Checkpatch just asked to remove the ```else``` condition that was unnecessary since there was a ```return``` instruction right before. 
It does not really matter anyway.